### PR TITLE
[postgres] add support for relation schemas

### PR DESCRIPTION
**Pull request by kostas0**
This change populates an additional tag for postgres relations containing the schema name. With this separate tag we can disambiguate metrics coming from same name tables but in separate schemas (e.g. staging vs production).

In addition we can now limit publishing to specific relation schemas via configuration.

Original issue #1496.

### CODE REVIEW

**Comment by remh** [comment]
Thanks Kostas ! 
Can you also update the postgres.yaml.example to explain how to use this feature and what parameters to use ?

**Comment by kostas0** [comment]
Added config syntax example.

File: conf.d/postgres.yaml.example [hunk 1]
```diff
@@ -18,6 +18,14 @@ instances:
 #    relations:
 #      - my_table
 #      - my_other_table
+#
+# By default all schemas are included. To track relations from specific schemas only, 
+# use the following syntax:
+#
+#    relations:
+#      - relation_name: another_table
+#        schemas: public prod
```
Comment (remh) [inline conf.d/postgres.yaml.example]: schemas should be a yaml list, instead of a space separated list

File: checks.d/postgres.py [hunk 2]
```diff
@@ -354,6 +358,25 @@ def _get_replication_metrics(self, key, db):
             metrics = self.replication_metrics.get(key)
         return metrics
 
+    def _build_relations_config(self, yc):
+        """Builds a dictionary from relations configuration while maintaining compatibility
+        """
+        config = {}
+        for x in yc:
```
Comment (remh) [inline checks.d/postgres.py]: You should use more explicit variable names

File: checks.d/postgres.py [hunk 3]
```diff
@@ -354,6 +358,25 @@ def _get_replication_metrics(self, key, db):
...
+                if isinstance(x, str):
+                    config[x] = {'relation_name': x, 'schemas': []}
+                elif isinstance(x, dict):
+                    name = x['relation_name']
+                    config[name] = {}
+                    config[name]['schemas'] = x['schemas'].split()
+                    config[name]['relation_name'] = name
+                else:
+                    self.log.warn('Unhandled relations config type: %s' % str(x))
+            except:
```
Comment (remh) [inline checks.d/postgres.py]: bare:
```python
except:
```
should almost always be avoided as they would catch  SystemExit, KeyboardInterrupt, and GeneratorExit as well which you don't want. See https://docs.python.org/2/howto/doanddont.html#except for more details.

You should catch the specific exceptions that can be expected, or if they are too many, you should do a:
```python
except Exception:
```

File: checks.d/postgres.py [hunk 4]
```diff
@@ -354,6 +358,25 @@ def _get_replication_metrics(self, key, db):
...
+                pass
```
Comment (remh) [inline checks.d/postgres.py]: Should we log something here ?

File: checks.d/postgres.py [hunk 5]
```diff
@@ -461,7 +497,7 @@ def _collect_stats(self, key, db, instance_tags, relations, custom_metrics):
                     else:
                         tags = [t for t in instance_tags]
 
-                    tags += ["%s:%s" % (d[0][1], d[1]) for d in zip(desc, row[:len(desc)])]
+                    tags += [("%s:%s" % (k,v)) for (k,v) in desc_map.items()]
```
Comment (remh) [inline checks.d/postgres.py]: You should use .iteritems() instead of .items() in python2

**Comment by kostas0** [comment]
Thanks for the suggestions, now updated, please take a look.

**Comment by remh** [comment]
Looks good to me!
 Thanks!

**Pull request merged**